extensions/api/keypair-mgmt-api/src/main/java/org/eclipse/edc/identityhub/api/keypair/v1/KeyPairResourceApiController.java

```diff
@@ -114,4 +120,25 @@ public void revokeKey(@PathParam("keyPairId") String id, KeyDescriptor newKey, @
                 .compose(u -> keyPairService.revokeKey(id, newKey))
                 .orElseThrow(exceptionMapper(KeyPairResource.class, id));
     }
+
+    /**
+     * Execute a participant-level operation after decoding its id.
+     *
+     * @param encoded  base64-encoded participant id.
+     * @param function participant level operation.
+     * @param <T>      operation return type.
+     * @return result of the operation execution.
+     */
+    private <T> T onParticipant(String encoded, Function<String, T> function) {
```

Comment:
instead of having this on every controller: could we move the `onParticipant` method to a central location? Either `spi` or some `core:util` module would fit the bill.

Reply:
and another thought: if `onParticipant` returned a `Result<String>`, we could elegantly `compose()` it together with other result-producing methods like the authorization etc.